### DISCUSSION TITLE
Сохранить порядок в рамках альбома при сортировке по дате релиза

### DISCUSSION THREAD

**Discussion by B1gV** [Chimildic/goofy#43] [Идея] [answered]
Приветствую!

Помогите, пожалуйста, написать код для плейлиста, состоящего из свежих релизов артистов из подписок, отсортированных по дате выхода, причем чтобы свежие релизы были сверху.

Мой код:

```
function TrueRadarSingle() {

let tracks = Source.getArtistsTracks({
    artist: {
        followed_include: true,
    },
    album: {
        groups: 'single',
        release_date: { sinceDays: 90, beforeDays: 0 },
    },
});

Filter.dedupTracks (tracks);
Order.sort (tracks, 'album.release_date', 'desc');

Playlist.saveWithReplace({
    id: '',
    public: false,
    tracks: tracks,
});

}
```

имеет критический недостаток: порядок треков в релизах меняется на обратный, это неприемлемо в ряде случаев, особенно в аналогичном варианте с альбомами, вместо синглов.

**Comment by amd64fox** [comment]
Если правильно понял проблему то вместо `'desc' `поставьте `'asc'` в `Order.sort`
То есть по сути менять нечего не надо просто удалить `'desc' ` тк `'asc'` идет по умолчанию и его не требуется прописывать.

**Reply by B1gV** [reply]
Тогда новые релизы будут внизу списка, не подходит.

**Reply by amd64fox** [reply]
тогда попробуйте оставить `'desc'` но добавить после строчку `Order.reverse(tracks);`

просто я проверить не могу сейчас, если не поможет то нужно разработчика ждать

**Reply by B1gV** [reply]
Ну и будет ровно как `asc`

**Reply by efremych** [reply]
А если вот так?

```js
Filter.dedupTracks (tracks);
Order.reverse(tracks);
Order.sort (tracks, 'album.release_date', 'desc');
```

**Comment by Chimildic** [comment, marked as answer]
[Обновите код](https://script.google.com/d/1DnC4H7yjqPV2unMZ_nmB-1bDSJT9wQUJ7Wq-ijF4Nc7Fl3qnbT0FkPSr/edit?usp=sharing), версию не менял (1.4.3)

**Reply by B1gV** [reply]
Все работает, спасибо огромное!
